### DISCUSSION TITLE
Microsoft.Identity.Client with C++/WinRT

### DISCUSSION THREAD

**Discussion by Deltaxel** [microsoft/react-native-windows#9230] [Q&A] [answered]
Hi,

I'm trying to create a native module for my C++/WinRT app in react-native. The objective is to leverage native feature for authentication and token acquisition.
The library I want to use is `Microsoft.Identity.Client`. However, it does not target "native". Which is why NuGet install keeps failing.

Is there a way to call this library anyway? Is there any sample showing similar behavior?

Thanks

**Comment by chrisglein** [comment, marked as answer]
There are some considerations for mixing C++ and C# that you can see here: https://microsoft.github.io/react-native-windows/docs/native-code-language-choice

There are a variety of auth modules for RN in general, but there aren't any here that are reporting Windows support: https://reactnative.directory/?search=auth
Some efforts have been made on those which may or may not be useful for you: https://github.com/auth0/react-native-auth0/pull/356
